geopandas/tests/test_dissolve.py

```diff
@@ -297,3 +297,21 @@ def test_dissolve_dropna_warn(nybb_polydf):
         UserWarning, match="dropna kwarg is not supported for pandas < 1.1.0"
     ):
         nybb_polydf.dissolve(dropna=False)
+
+
+def test_dissolve_multi_agg(nybb_polydf, merged_shapes):
+
+    merged_shapes[("BoroCode", "min")] = [3, 1]
+    merged_shapes[("BoroCode", "max")] = [5, 2]
+    merged_shapes[("BoroName", "count")] = [3, 2]
+
+    with pytest.warns(None) as record:
+        test = nybb_polydf.dissolve(
+            by="manhattan_bronx",
+            aggfunc={
+                "BoroCode": ["min", "max"],
+                "BoroName": "count",
+            },
+        )
+    assert test.geom_almost_equals(merged_shapes).all()
```

Comment:
```suggestion
    assert_geodataframe_equal(test, merged_shapes)
```

This should ensure it's actually testing the full dataframe, and not only the geometry column.

You might still have to import this function from `geopandas.testing` (it seems the other tests in this file are not yet using it)

Reply:
Ah, thanks. Sure, we want to test particularly that we have a flat index